src/main.ts

```diff
@@ -0,0 +1,256 @@
+    }
+  } else {
+    lastStatus = await getStatusFromGithub();
+  }
+
+  return lastStatus.trim();
+}
+
+/**
+ * Prepare slack notification
+ *
+ * @param message contained in the Slack notification
+ * @param status current status to notify
+ * @returns the Slack message body
+ */
+async function prepareSlackNotification(message: string, status: string) {
+  const {
+    runId,
+    workflow,
+    actor,
+    serverUrl,
+    eventName: event,
+    repo: { owner },
+  } = context;
+  const color = status === 'success' ? 'good' : 'danger';
+
+  const headRef = await getHeadRef();
+
+  const messageBody = {
+    icon_emoji: ':bangbang:', // User icon, you can also use custom icons here
+    attachments: [
+      {
+        // this defines the attachment block, allows for better layout usage
+        color: `${color}`, // color of the attachments sidebar.
+        author_name: `${actor}`,
+        author_link: `${serverUrl}/${actor}`,
+        author_icon: `${serverUrl}/${actor}.png?size=32`,
+        fields: [
+          // actual fields
+          {
+            title: 'Repository',
+            value: `${repository}`,
+            short: true,
+          },
+          {
+            title: 'Branch',
+            value: `${headRef}`,
+            short: true,
+          },
+          {
+            title: 'Action URL',
+            value: `<${serverUrl}/${owner}/${repository}/actions/runs/${runId}|${workflow}>`,
+            short: true,
+          },
+          {
+            title: 'Event',
+            value: `${event}`,
+            short: true,
+          },
+          {
+            title: `${workflow} workflow ${status}`,
+            value: `${message}`, // Custom value
+            short: false, // long fields will be full width
+          },
+        ],
+      },
+    ],
+  };
+  return messageBody;
+}
+
+/**
+ * Handles the actual sending request.
+ * We're turning the https.request into a promise here for convenience
+ *
+ * @param webhookURL URL of the Slack channel Webhook
+ * @param messageBody Payload to send to Slack
+ */
+async function sendSlackMessage(
+  webhookURL: string,
+  messageBody: Record<string, any>,
+) {
+  core.info(`Message body: ${JSON.stringify(messageBody)}`);
+
+  await got.post(webhookURL, {
+    json: messageBody,
+  });
+}
+
+/**
+ * Logs an error and fails the Github Action
+ *
+ * @param err Any possible errors
+ */
+function handleError(err: Error): void {
+  console.error(err);
+  core.setFailed(err.message);
+}
+
+/**
+ * Action run pipeline
+ */
+async function pipeline() {
+  const lastStatus = await getLastRunStatus();
+  const currentStatus = core.getInput(Inputs.CurrentStatus);
+  const webhookUrl = core.getInput(Inputs.SlackWebhook);
+
+  core.info(`Last run status: ${lastStatus}`);
+  core.info(`Current run status: ${currentStatus}`);
+
+  if (currentStatus !== 'success' && currentStatus !== 'failure') {
+    core.setFailed('Wrong current status value');
+    return;
+  }
+
+  let URLTest;
```

Comment:
Nit: Since this variable is going to be an instance of the URL class, it should be lowercase.

```suggestion
  let url;
```